python/mlc_chat/cli/convert_weight.py

```diff
@@ -0,0 +1,132 @@
+"""Command line entrypoint of weight conversion."""
+import argparse
+import logging
+from pathlib import Path
+from typing import Union
+
+from ..support.auto_config import detect_config, detect_model_type
+from ..support.auto_target import detect_target_and_host
+
+from tvm.contrib import tvmjs
+from mlc_chat.compiler import MODELS, QUANTIZATION
+
+from mlc_chat.compiler.parameter import HuggingFaceLoader
+
+logging.basicConfig(
+    level=logging.INFO,
+    style="{",
+    datefmt="%Y-%m-%d %H:%M:%S",
+    format="[{asctime}] {levelname} {filename}:{lineno}: {message}",
+)
+
+
+def main():
+    """Parse command line argumennts and apply quantization."""
+
+    def _parse_config(path: Union[str, Path]) -> Path:
+        try:
+            return detect_config(path)
+        except ValueError as err:
+            raise argparse.ArgumentTypeError(f"No valid config.json in: {path}. Error: {err}")
+
+    def _parse_source(path: Union[str, Path], config_path: Path) -> Path:
+        if path == "auto":
+            return config_path.parent
+        path = Path(path)
+        if not path.is_dir():
+            raise argparse.ArgumentTypeError(f"Directory does not exist: {path}")
+        return path
+
+    def _parse_output(path: Union[str, Path]) -> Path:
+        path = Path(path)
+        if not path.is_dir():
+            path.mkdir(parents=True, exist_ok=True)
+        return path
+
+    parser = argparse.ArgumentParser("MLC AutoLLM Quantization Framework")
+    parser.add_argument(
+        "--config",
+        type=_parse_config,
+        required=True,
+        help="Path to config.json file or to the directory that contains config.json, which is "
+        "a HuggingFace standard that defines model architecture, for example, "
+        "https://huggingface.co/codellama/CodeLlama-7b-Instruct-hf/blob/main/config.json",
+    )
+    parser.add_argument(
+        "--source",
+        type=str,
+        required=False,
+        default="auto",
+        help="The path to original model weight, infer from `config` if missing",
+    )
+    parser.add_argument(
+        "--source-format",
+        type=str,
+        required=False,
+        # todo support auto
+        choices=["huggingface-torch", "huggingface-safetensor"],  # "auto"
+        default="huggingface-safetensor",  # "auto"
+        help="The format of source model weight, infer from `config` if missing",
+    )
```

Comment:
You may want to use weight format detection from @LeshengJin: https://github.com/mlc-ai/mlc-llm/pull/1088